pkg/controller/certificates/controller.go

```diff
@@ -181,6 +187,51 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	}
 	csr := obj.(*certificates.CertificateSigningRequest)
 
+	func() {
+		if !cc.approveAllKubeletCSRs ||
```

Comment:
What error are you seeing?

> why is this wrapped in a lambda?

returns offer simpler control flow. It should probably be extracted to a seperate function. It could just as easily be nested ifs or a goto